From 2046618f6f989f148baa254404cb4d4437fb8a18 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Fri, 7 Aug 2026 12:53:12 +0200
Subject: [PATCH] Clear temporary files on close

Without this, all temporary files remain on the system for each pasted image.
---
 src/main.py    | 10 ++++++++++
 src/pasting.py |  7 +++++++
 src/window.py  |  7 +++++++
 3 files changed, 24 insertions(+)

diff --git a/src/main.py b/src/main.py
index d396f89..1734f9f 100644
--- a/src/main.py
+++ b/src/main.py
@@ -72,6 +72,7 @@ class LetterpressApplication(Adw.Application):
             "open-output", self.__open_output, param=GLib.VariantType("s")
         )
         self.file = None
+        self.tmpfile = None
 
     def do_activate(self):
         """Called when the application is activated.
@@ -143,6 +144,15 @@ class LetterpressApplication(Adw.Application):
         self.activate()
         return 0
 
+    def do_shutdown(self):
+        if self.tmpfile:
+            try:
+                self.tmpfile.delete()
+            except GLib.Error:
+                pass
+
+        Gtk.Application.do_shutdown(self)
+
     def __on_about_action(self, *args):
         """Callback for the app.about action."""
         about = Adw.AboutDialog.new_from_appdata(
diff --git a/src/pasting.py b/src/pasting.py
index 36c9122..57cc0de 100644
--- a/src/pasting.py
+++ b/src/pasting.py
@@ -42,6 +42,13 @@ class Paster:
         try:
             paste_as_texture = clipboard.read_texture_finish(result)
             file = texture_to_file.to_file(paste_as_texture)
+            app = self.parent_window.get_application()
+            if app.tmpfile:
+                try:
+                    app.tmpfile.delete()
+                except GLib.Error:
+                    pass
+            app.tmpfile = file
             self.callback(file)
 
         except:
diff --git a/src/window.py b/src/window.py
index 30245dd..5aab3ef 100644
--- a/src/window.py
+++ b/src/window.py
@@ -137,6 +137,13 @@ class LetterpressWindow(Adw.ApplicationWindow):
 
         try:
             file = texture_to_file.to_file(drop)
+            app = self.get_application()
+            if app.tmpfile:
+                try:
+                    app.tmpfile.delete()
+                except GLib.Error:
+                    pass
+            app.tmpfile = file
             self.convert_image(file)
         except:
             toast = Adw.Toast.new(_("Dropped item is not a valid image"))
-- 
2.55.0

